tests/test_downloadermiddleware_retry.py

```diff
@@ -110,6 +110,29 @@ def test_twistederrors(self):
             == 2
         )
 
+    def test_exception_to_retry_added(self):
+        # Don't retry
+        exc = ValueError
+        req = Request(f"http://www.scrapytest.org/{exc.__name__}")
+        req = self.mw.process_exception(req, exc("foo"), self.spider)
+        self.assertEqual(req, None)
+        # Retry
+        self.mw.exceptions_to_retry = self.mw.exceptions_to_retry + (exc,)
+        req = Request(f"http://www.scrapytest.org/{exc.__name__}")
+        self._test_retry_exception(req, exc("foo"))
```

Comment:
It would be best to test adding it the way users should add it, e.g. (untested)

```python
from scrapy.settings.default_settings import RETRY_EXCEPTIONS
settings = {
    "RETRY_EXCEPTIONS": RETRY_EXCEPTIONS + (ValueError,),
}
crawler = get_crawler(Spider, settings=settings)
mw = RetryMiddleware.from_crawler(crawler)
req = Request(f"http://www.scrapytest.org/{exc.__name__}")
self._test_retry_exception(req, exc("foo"), mw)  # Modify to accept an optional mw
```